### PR TITLE
Feature/#3119 adjustable dm priority for huts

### LINKED ISSUE #3119: Suggested Feature: Adjustable DM priority for huts.
Being able to adjust the building delivery/collection priority would allow a player to put emphasis on "specializations" and would also give the option to "disable" a building once they move past that point or need to divert DM's to other areas they want to focus on by allowing a player to set the DM priority to 0 or off or something similar. Some examples would be if you consider Fisherman to be early game and Farmer to be mid-end game you can turn off the fisherman once you advance to farmer. Same logic with Baker>Cook. 

Credit: LordNavarre#9565(161237577519726593) on the discord who didn't have a Git account to post this with.

### PR DESCRIPTION
**Pull request by nojo121**
Closes #3119 

# Changes proposed in this pull request:
- Added Manual Delivery Settings

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/client/gui/AbstractWindowWorkerBuilding.java [hunk 1]
```diff
@@ -1,163 +1,219 @@
...
+        super.registerButton(BUTTON_CRAFTING, this::craftingClicked);
+        super.registerButton(BUTTON_RECIPES_LIST, this::recipeListClicked);
+        super.registerButton(BUTTON_DP_UP, this::deliveryPrioUp);
+        super.registerButton(BUTTON_DP_DOWN, this::deliveryPrioDown);
+        super.registerButton(BUTTON_DP_STATE, this::changeDPState);
+    }
+
+    private void deliveryPrioUp()
+    {
+        if (prio != 10)prio++;
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/client/gui/AbstractWindowWorkerBuilding.java:119]: please { } around it. (same for the other ifs)

File: src/main/java/com/minecolonies/coremod/colony/buildings/views/AbstractBuildingView.java [hunk 2]
```diff
@@ -1,422 +1,434 @@
...
+            return new TextComponentString("");
+        }
+    }
+
+    public int getBuildingDmPrio()
+    {
+        return buildingDmPrio;
+    }
+
+    public boolean getBuildingDmPrioState()
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/colony/buildings/views/AbstractBuildingView.java:440]: javadoc

File: src/main/java/com/minecolonies/coremod/network/messages/ChangeDeliveryPriorityStateMessage.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            //Verify player has permission to change this huts settings
+            if (!colony.getPermissions().hasPermission(player, Action.MANAGE_HUTS))
+            {
+                return;
+            }
+            final AbstractBuilding building = colony.getBuildingManager().getBuilding(message.buildingId);
+
+            if(building instanceof AbstractBuildingWorker)
+            {
+                MineColonies.getLogger().info(building.getPriorityState());
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/network/messages/ChangeDeliveryPriorityStateMessage.java]: I think we can remove the logger here now =)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java [hunk 4]
```diff
@@ -1,705 +1,709 @@
...
+            if (ItemStackUtils.isEmpty(stack))
+            {
+                continue;
+            }
+
+            final ItemStack insertionResultStack;
+
+            if (tileEntityColonyBuilding.getBuilding() instanceof AbstractBuildingWorker)
+            {
+                if (!building.getPriorityState())
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java]: Again here {}

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java [hunk 5]
```diff
@@ -1,705 +1,709 @@
...
+            countWeight += building.getPickUpPriority();
+            if (countWeight >= r)
+            {
+                //Don't let any other dman pick up for now.
+                building.setBeingGathered(true);
+                return building.getID();
+            }
+            else
+            {
+                if (!building.getPriorityState())
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java]: Again here {}

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java [hunk 6]
```diff
@@ -1,705 +1,709 @@
...
+                if (hasGathered)
+                {
+                    job.setReturning(true);
+                    this.hasGathered = false;
+                    if (!building.getPriorityState())
+                        building.alterPickUpPriority(1);
+                }
+                else
+                {
+                    if (!building.getPriorityState())
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java]: Again here {}

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java [hunk 7]
```diff
@@ -1,705 +1,709 @@
...
+                this.currentSlot = 0;
+                building.setBeingGathered(false);
+
+                gatherCount ++;
+
+                if (hasGathered)
+                {
+                    job.setReturning(true);
+                    this.hasGathered = false;
+                    if (!building.getPriorityState())
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java]: Again here {}

File: src/main/java/com/minecolonies/coremod/network/messages/ChangeDeliveryPriorityMessage.java [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+                if (message.up)
+                {
+                    building.alterPickUpPriority(1);
+                    building.markDirty();
+                }
+                else
+                {
+                    building.alterPickUpPriority(-1);
+                    building.markDirty();
+
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/network/messages/ChangeDeliveryPriorityMessage.java]: Remove empty line.

**Review by marchermans — DISMISSED** [review]
Some small changes that relate to formatting. As soon as you processed those we are good.
For the internals: :shipit:

File: src/main/java/com/minecolonies/coremod/network/messages/ChangeDeliveryPriorityStateMessage.java (hunk 3 above)
Comment (nojo121) [reply src/main/java/com/minecolonies/coremod/network/messages/ChangeDeliveryPriorityStateMessage.java]: woops

**Pull request merged**